consensus/tendermint/backend/engine_test.go

```diff
@@ -453,22 +458,253 @@ func TestClose(t *testing.T) {
 		b := &Backend{}
 
 		err := b.Close()
-		if err != ErrStoppedEngine {
-			t.Fatalf("expected %v, got %v", ErrStoppedEngine, err)
-		}
+		assertNotCoreStarted(t, b)
+		assertError(t, ErrStoppedEngine, err)
 	})
 
 	t.Run("engine is running, no errors", func(t *testing.T) {
+		ctrl := gomock.NewController(t)
+		defer ctrl.Finish()
+
+		tendermintC := tendermintCore.NewMockTendermint(ctrl)
+		tendermintC.EXPECT().Stop().MaxTimes(1)
+
 		b := &Backend{
+			core:        tendermintC,
 			coreStarted: true,
 			stopped:     make(chan struct{}),
 		}
 
 		err := b.Close()
-		if err != nil {
-			t.Fatalf("expected <nil>, got %v", err)
+		assertNotCoreStarted(t, b)
+		assertNilError(t, err)
+	})
+
+	t.Run("engine is running, stopped twice", func(t *testing.T) {
+		ctrl := gomock.NewController(t)
+		defer ctrl.Finish()
+
+		tendermintC := tendermintCore.NewMockTendermint(ctrl)
+		tendermintC.EXPECT().Stop().MaxTimes(1)
+
+		b := &Backend{
+			core:        tendermintC,
+			coreStarted: true,
+			stopped:     make(chan struct{}),
 		}
+
+		err := b.Close()
+		assertNotCoreStarted(t, b)
+		assertNilError(t, err)
+
+		err = b.Close()
+		assertNotCoreStarted(t, b)
+		assertError(t, ErrStoppedEngine, err)
+	})
+
+	t.Run("engine is running, stopped multiple times", func(t *testing.T) {
+		ctrl := gomock.NewController(t)
+		defer ctrl.Finish()
+
+		tendermintC := tendermintCore.NewMockTendermint(ctrl)
+		tendermintC.EXPECT().Stop().MaxTimes(1)
+
+		b := &Backend{
+			core:        tendermintC,
+			coreStarted: true,
+			stopped:     make(chan struct{}),
+		}
+
+		var wg sync.WaitGroup
+		stop := 10
+		errC := make(chan error, stop)
+
+		for i := 0; i < stop; i++ {
+			wg.Add(1)
+
+			go func(wg *sync.WaitGroup) {
+				defer wg.Done()
+
+				err := b.Close()
+				errC <- err
```

Comment:
```suggestion
				errC <- b.Close()
```

There's no need to create a variable here.

Reply:
Ok I will update